src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -260,6 +282,50 @@ private async Task<Document> InlineTemporaryAsync(Document document, VariableDec
             return updatedDocument;
         }
 
+        private static bool MayContainSideEffects(SyntaxNode expression)
+        {
+            // Checks to see if inlining the temporary variable may change the code's semantics. 
```

Comment:
comment that this isn't meant to be an exhaustive/correct check, more like a heuristic for obvious things we want to call attention to.